pkg/collector/tcp.go

```diff
@@ -47,18 +45,15 @@ func (cp *CollectingProcess) startTCPServer() {
 				klog.Errorf("Cannot start collecting process on %s: %v", cp.address.String(), err)
 				return
 			}
-			wg.Add(1)
-			go cp.handleTCPClient(conn, &wg)
+			go cp.handleTCPClient(conn)
 		}
 	}()
 	<-cp.stopChan
 	// close all connections
 	cp.closeAllClients()
-	wg.Wait()
```

Comment:
As stopChan is there, is waitGroup redundant?

Reply:
Yes